src/main.c

```diff
@@ -592,6 +594,7 @@ static bool check_faults(data *d) {
             }
         } else {
             d->fault_switch_timer = d->current_time;
+            d->motor.last_erpm_sign = d->motor.erpm_sign;
```

Comment:
Modifying the `MotorData` structure outside of the `motor_data_` functions goes against the concept of the modularization (which is, any work with the module data happens inside the module, outside of it, you only read them).

I'm also not sure there's more than meets the eye to the surgical precision with which you placed the assignment in two places of the `check_faults()` function :grinning:. But the fact it's two places is bad in itself, and this function really needs stuff removed, not added (it needs a refactor).

Instead, please set `last_erpm_sign` inside `motor_data_update()`. You'll have to pass whatever data you need for the condition. Seems to me you need `d->state.state` and check it's equal to `STATE_RUNNING` and keep updating the var for as long as that holds true. But I might have missed a detail in how you placed those assignments, it's not entirely obvious (which is the point I'm trying to make).

I'd call the variable `erpm_last_running_sign`.